radio/sdcard/horus/SCRIPTS/WIZARD/heli/wizard.lua

```diff
@@ -0,0 +1,592 @@
+  else
+    model.setCurve(1,{name="TC1",y={85 ,85,85}})
+  end
+
+  --Curve FM2
+  if StyleFields[1][5]>=0 and CurveFields[3][5]==0 then
+    model.setCurve(2,{name="Tc2",y={70, 60, 70}})
+  elseif StyleFields[1][5]==1 and CurveFields[3][5]==0 then
+    model.setCurve(2,{name="TC2",y={85, 70, 85}})
+  elseif StyleFields[1][5]==2 and CurveFields[3][5]==0 then
+    model.setCurve(2,{name="TC2",y={100, 90, 100}})
+  elseif StyleFields[1][5]==0 and CurveFields[3][5]==1 then
+    model.setCurve(2,{name="TC2",y={75 ,75,75}})
+  elseif StyleFields[1][5]==1 and CurveFields[3][5]==1 then
+    model.setCurve(2,{name="TC2",y={85 ,85, 85}})
+  else
+    model.setCurve(2,{name="TC2",y={95 ,95, 95}})
+  end
+
+  --Curve TH Hold
+  model.setCurve(3,{name="THD",y={-100,-100,-100}})
+
+  -- Throttle
+  model.insertMix(ThrFields[1][5], 0,{name="Th0",weight=100,curveType=3,curveValue=1})
+  model.insertMix(ThrFields[1][5], 1,{name="Th1",weight=100,switch=tUp,multiplex=2,curveType=3,curveValue=2})
+  model.insertMix(ThrFields[1][5], 2,{name="Th2",weight=100,switch=tUp-1,multiplex=2,curveType=3,curveValue=3})
+  model.insertMix(ThrFields[1][5], 3,{name="Hld",weight=100,offset=-15,switch=hold+1,multiplex=2,curveType=3,curveValue=4})
+
+  -- Ail
+  if TypeFields[1][5] == 0 then
+    model.insertMix(AilerFields[1][5], 0,{name="Ail",weight=100})
+  else
+    model.insertMix(AilerFields[1][5], 0,{source=102,name="Ail",weight=100})
+  end
+
+  -- Elev
+  if TypeFields[1][5] == 0 then
+    model.insertMix(EleFields[1][5], 0,{name="Ele",weight=100})
+  else
+    model.insertMix(EleFields[1][5], 0,{source=101,name="Ele",weight=100})
+  end
+
+  -- Rudder
+  model.insertMix(RudFields[1][5], 0,{name="Rud",weight=100})
+
+  -- Gyro
+  if TypeFields[1][5] == 0 then
+    model.insertMix(4, 0,{source=100,name="Gyr",weight=25})
+  else
+    model.insertMix( 4, 0,{source=100,name="HH",weight=25})
+    model.insertMix( 4, 1,{source=100,name="Rat",weight=-25,switch=gyRate+1,multiplex=2})
+  end
+
+  -- Pitch
+  if TypeFields[1][5] == 0 then
+    model.insertMix(5, 0,{source=89,name="Pch",weight=100})
+  else
+    model.insertMix(5, 0,{source=103,name="Pch",weight=100})
+  end
+
+  --Set Swash Parameters
+  if TypeFields[1][5]==1 and TypeFields[2][5]==0 then
+    model.swashRingData(0,0,{type="1",collectiveSource=89,aileronSource=90,elevatorSource=88,collectiveWeight=60,aileronWeight=60,elevatorWeight=60})
+  elseif TypeFields[2][5]==1 then
+    model.swashRingData(0,0,{type="2",collectiveSource=89,aileronSource=90,elevatorSource=88,collectiveWeight=60,aileronWeight=60,elevatorWeight=60})
+  elseif TypeFields[2][5]==2 then
+    model.swashRingData(0,0,{type="3",collectiveSource=89,aileronSource=90,elevatorSource=88,collectiveWeight=40,aileronWeight=40,elevatorWeight=60})
+  elseif TypeFields[2][5]==2 then
+    model.swashRingData(0,0,{type="4",collectiveSource=89,aileronSource=90,elevatorSource=88,collectiveWeight=35,aileronWeight=35,elevatorWeight=60})
+  else
```

Comment:
Something is missing or too much

Reply:
Without the else end (do nothing) , it always adds info to the heli screen.  This information is not needed for FBL helicopters, this screen pertains to flybar helicopters.

instead of a "do nothing" at the end I can move it to the beginning

if TypeFields[1][5]==0 then 
end
else
if TypeFields[2][5]==0 then 
...
else
model.swashRingData(0,0,{type="4" ....
end

It changes the location of the "do nothing" so that the command ends with a flybar choice.

Which method is preferred?

Reply:
Changed the location of the do nothing  to the front of the command - unwanted results.
With an "else statement" at the end, the swash parameters are always filled in regardless of helicopter type - flybarless or flybar.  (Since it didn't meet any of the other requirements it chose the last one).

Item reverted back to original setting with do nothing at the end.